### ISSUE TITLE
Support for Debug and Release versions

### ISSUE THREAD

**Issue by trullock** [OpenWrap/openwrap#231]
Would it be possible to allow wraps to contain debug and release versions of libraries, such that the project references the one with the same configuration?

This way when I'm debugging I can have debug symbols for libs, but when releasing it uses the optimised (release) version?

**Comment by serialseb** [comment]
I'm not sure how we would fit that. On one hand, I can see why it'd be good, on the other hand I'm not entirely sure just compiling to another package name and using overrides wouldn't already solve this.

The build command supports -debug and -release in 2.0 to build packages in different modes, and also ought to support -flavour where each flavour contains a different set of files for a single build.

We could simply build both in debug and release mode and then ship both assemblies, maybe splitting them in different packages using flavours, but that poses a problem with dependencies. How do you envision using debug references? Per-build (if building in debug, use debug assemblies, if in release, build release assemblies)?

Need to have a long think about this. Any pointers to how other (non-.net) package managers solve that problem?

**Comment by trullock** [comment]
Yeah I realised i could have two packages and switch with overrides, but thats manual, was hoping for something more automatic.

Yeah, one package could contain:

/bin-net40/debug/my.dll
/bin-net40/release/my.dll

and then whatever configuration im compiling my dependee assembly, use the relevant assembly from the wrap.

Im not familiar with flavours :s

**Comment by serialseb** [comment]
Only issue really is that configurations dont really mean much for OpenWrap.

Let be think about it a little bit.
